skyportal/handlers/api/annotation.py

```diff
@@ -30,26 +49,65 @@ def get(self, annotation_id):
               application/json:
                 schema: Error
         """
-        annotation = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, raise_if_none=True
-        )
+
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid annotation ID. ")
```

Comment:
This should return "Annotation not found" with `status=404`.

Reply:
I'm not sure. I think this catches cases where the ID is None or not an integer. Seems to me "Annotation not found" is more relevant when an integer is given but does not fit any row in the database.

Reply:
OK, that makes sense.  In that case, specific guidance would be helpful to the user: "Integer expected for annotation ID" or similar.